src/Game.ts

```diff
@@ -100,7 +113,13 @@ class Game {
             if (!field?.piece) return;
 
             if (this.turn === field.piece.side) {
-                const possibleMoves = field.piece.findLegalMoves(this.board, field);
+                const possibleMoves = field.piece.findLegalMoves(this.board, field)
+                .filter(move => {
+                    let newField = this.board.getField(parseInt(move[0]), parseInt(move[2]));
+                    const mirror = this.canMove(field, newField);
+                    return !this.isCheck(mirror);
```

Comment:
Tutaj powinno być tylko wyfiltrowanie z tej metody canMove i jeżeli ona zwróci true to sie nic nie dzieje a jeżeli false to nie ma mozliwości tam ruchu